numbat/src/unit.rs

```diff
@@ -42,6 +42,18 @@ impl UnitIdentifier {
         matches!(self.kind, UnitKind::Base)
     }
 
+    pub fn unit_and_factor(&self) -> BaseUnitAndFactor {
```

Comment:
The idea behind this function is an analog to `base_unit_and_factor` except that it doesn't convert it into the base unit.